instrumentation/github.com/aws/aws-lambda-go/otellambda/xrayconfig/xrayconfig.go

```diff
@@ -35,48 +34,39 @@ func xrayEventToCarrier([]byte) propagation.TextMapCarrier {
 	return propagation.HeaderCarrier{"X-Amzn-Trace-Id": []string{xrayTraceID}}
 }
 
-type asyncSafeFlusher struct {
-	tp *sdktrace.TracerProvider
-}
-
-func (f asyncSafeFlusher) ForceFlush(ctx context.Context) error {
-	// yield processor to attempt to ensure all spans have
-	// been consumed and are ready to be flushed
-	// - see https://github.com/open-telemetry/opentelemetry-go/issues/2080
-	// to be removed upon resolution of above issue
-	runtime.Gosched()
-
-	return f.tp.ForceFlush(ctx)
-}
-
-// tracerProviderAndFlusher returns a list of otellambda.Option(s) to
-// enable using a TracerProvider configured for AWS XRay via a collector
-// and an otellambda.Flusher to flush this TracerProvider.
-// tracerProviderAndFlusher is not exported because it should not be used
-// without the provided EventToCarrier function and XRay Propagator
-func tracerProviderAndFlusher() ([]otellambda.Option, error) {
-	ctx := context.Background()
-
-	// Do not need transport security in Lambda because collector
-	// runs locally in Lambda execution environment
+// NewTracerProvider returns a TracerProvider configured with exporter,
+// id generator and lambda resource detector to send trace data to AWS X-Ray via Collector
+func NewTracerProvider(ctx context.Context) (*sdktrace.TracerProvider, error) {
 	exp, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure())
 	if err != nil {
-		return []otellambda.Option{}, err
+		errorLogger.Println("failed to create exporter: ", err)
+		return nil, err
 	}
 
 	detector := lambdadetector.NewResourceDetector()
-	res, err := detector.Detect(ctx)
+	resource, err := detector.Detect(ctx)
 	if err != nil {
-		return []otellambda.Option{}, err
+		errorLogger.Println("failed to detect lambda resources: ", err)
+		return nil, err
 	}
 
-	tp := sdktrace.NewTracerProvider(
+	return sdktrace.NewTracerProvider(
 		sdktrace.WithBatcher(exp),
 		sdktrace.WithIDGenerator(xray.NewIDGenerator()),
-		sdktrace.WithResource(res),
-	)
+		sdktrace.WithResource(resource),
+	), nil
+}
+
+// TracerProvider returns an otellambda.Option(s) to
+// enable using a TracerProvider configured for AWS XRay via a collector
+func TracerProvider(tp *sdktrace.TracerProvider) otellambda.Option {
```

Comment:
We have followed that pattern in `config.go` file in `otellambda` package. I think for `xrayconfig.go` we have followed naming without `With` keyword. See [here](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/github.com/aws/aws-lambda-go/otellambda/xrayconfig/xrayconfig.go#L91) for `Propagator` and `EventToCarrier` method name for reference.

Reply:
I understand that other functions in the `xrayconfig` package were not adhering to the convention.  I now think that to be a mistake that we should fix at the earliest opportunity, a.k.a., now.

Reply:
I think it would be confusing because then public methods in `otellambda` packages would have the same name as public methods in `xrayconfig` packages and would confuse users but I'm open do it like the way you mentioned but just posting this as a concern.

Reply:
We should simply remove the superfluous functions.  The `TracerProvider()` and `Flusher()` functions provide no value as they merely pass through to the functions in the instrumentation package.  `WithRecommendedOptions()` can call the `Option`-generating functions in the instrumentation package directly.  `Propagator()` and `EventToCarrier()` should be renamed with the `With` prefix.